### PR TITLE
Add fish shell completions for lncli

### PR DESCRIPTION
**Pull request by torkelrogstad**
Add basic fish shell completions. Includes all global options
(with approriate available values) and all available subcommands
parsed from the output of the help command.

Does not add completions per subcommand.

### CODE REVIEW

**Comment by torkelrogstad** [comment]
To test: 

1) Install `fish`: https://fishshell.com/
  `sudo apt install fish` on Ubuntu
2) move `contrib/lncli.fish` to `$HOME/.config/fish/completions`
3) Launch fish shell, by simply doing `fish`
4) Type `lncli <TAB>` and verify that available subcommands gets suggested
5) Type `lncli - <TAB>` and verify that available options gets suggested
6) Type `lncli --chain <TAB>` or `lncli -c <TAB>` and verify that `bitcoin` and `litecoin` gets suggested
7) Type `lncli --network <TAB>` or `lncli -n <TAB>` and verify that `mainnet`, `regtest`, `simnet` and `testnet` gets suggested

File: contrib/lncli.fish [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+complete -c lncli -d 'the network lnd is running on' -l network -s n -x -a "mainnet testnet regtest simnet"
+complete -c lncli -d 'the chain lnd is running on' -l chain -s c -x -a "bitcoin litecoin"
+complete -c lncli -d 'path to TLS certificate' -l tlscertpath -x
+complete -c lncli -d "path to lnd's base directory" -l lnddir -x
+complete -c lncli -d "host:port of ln daemon" -l rpcserver -xf
+
+# get the list of all available subcommands
+# in the help section, all subcommands start 
+# with 4 spaces
+set commands (lncli help | grep '^    ' | grep -v help | sed -e 's/^[[:space:]]*//' | cut -f 1 -d " ")
```
Comment (guggero) [inline contrib/lncli.fish]: Maybe we could use the same `awk` command here that is already used for bash completion?
`lncli help 2>/dev/null | awk '$1 ~ /^[a-z]/ { print $1; }'`
That might be a bit more resilient against changes in how the help screen is formatted.

**Comment by dchirikov** [comment]
Sorry to interrupt, there is another PR from me about autocompletion. #2542

**Comment by Roasbeef** [comment]
Replaced by #4177.

**Pull request closed without merging**